### PR TITLE
TCOMP-326: Adds JavaScript client

### PR DESCRIPTION
**Pull request by talend-glorieux**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
There are no implementations of the component library web API.

**What is the new behavior?**
Now, there is one.

**Other information**:

Adds basic JavaScript client to the components web service.

That implementation is a rough draft based on the latest components web API specifications.
This is still a work in progress that will need to be improved by adding tests, examples and more. 
It's a conversation opener. Any idea/contribution is welcome.

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/363/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/363/)

**Comment by gonchar-ivan** [comment]
"components-" prefix is used for java Talend components projects developed using Component Framework.
Please, don't use this prefix for JavaScript client.

Also as I commented in Jira
https://jira.talendforge.org/browse/TCOMP-326

I'm not sure this code should go to TCOMP.

**Comment by RyanSkraba** [comment]
Hello!  We have a proposed re-organisation of the repo at: https://jira.talendforge.org/browse/TCOMP-249

If you're making a new module, it should be in the "re-org" directory from the start!  (I don't think we're going to move existing modules until after the 6.3.1 release, to prevent extra work for ongoing PR).

The REST api is in : `/services/components-api-service-rest`

What do you think about `/services/components-rest-client-js`?

**Comment by sgandon** [comment]
Also is it possible to have test running for this js client implem ?

**Comment by talend-glorieux** [comment]
I will move the client to the proper folder. 
@sgandon tests are on there way ;)

File: components-javascript-client/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: node_js
+cache:
+  directories:
+    - node_modules
+notifications:
+  email: false
+node_js:
+  - '4'
```
Comment (talend-glorieux) [inline components-javascript-client/.travis.yml]: Yep that might be it. Will update.

File: components-javascript-client/src/tcomp.js [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+		ValidateDatastoreProperties,
+		CheckDataStoreConnection,
+		DatasetProperties,
+		ValidateDatasetProperties,
+		CheckDatasetConnection,
+		DatasetSchema,
+		DatasetContent,
+		Transformers,
+		ValidateProperties,
+	};
```
Comment (talend-glorieux) [inline components-javascript-client/src/tcomp.js]: I'm not a big fan of putting usage before declaration. @Talend/frontend  What do you think ?

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/387/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/387/)

**Comment by sgandon** [comment]
How do you intend to build and test this library ?
We have a global build of the repo with maven, do you intend to hook into this build ?
I see some *travis* file how is it being used ?  this may require some more README information.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/399/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/399/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/402/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/402/)

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/403/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/403/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/404/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/404/)

**Comment by talend-glorieux** [comment]
I'll make a pull request to add the build/tests steps to the Jenkins configuration and add a README file.

**Comment by talend-glorieux** [comment]
Quick question on the folder naming. Can I drop the `rest` part of the name (components-rest-client-js) since the API is clearly not restful?

**Pull request merged**